ledger/ledger-api-domain/src/main/scala/com/digitalasset/ledger/api/ListUsersFilter.scala

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api
+
+import com.daml.ledger.api.domain.IdentityProviderId
+
+sealed trait ListUsersFilter
```

Comment:
I somehow dont like `Constraint` as it just a filter which might pass or filter stuff out. Constraint would the one which is currently called `ByIdentityProviderId`. wdyt?

Reply:
Not sure about the 'Constraint' either but the point I'm trying to make is to put focus on `IdentityProviderId` rather than on `ListUsers`..

I don't want to bike shed on this, so I'm fine with whatever you'll end up doing :)